### ISSUE TITLE
XSS char encoding like in Mantra

### ISSUE THREAD

**Issue by balsamiqFlorian** [zaproxy/zaproxy#2222]
Hi zap team,

Thanks for the great tool that helped me to detect some of the security flaws that were around.

In the encoder I was missing the html encoding with char codes. It would would be great if you could add it. Because filters that forget to decode the html and just look for text blocks might miss char encodings.

![mantra](https://cloud.githubusercontent.com/assets/3863260/12784065/3823ca2a-ca85-11e5-8a3e-4c36e4b15296.png)

It would be cool if you could add it so I could one tool less.

**Comment by psiinon** [comment]
So you'd want to see:
`<script>`
encoded as
`&lt;&#115;&#99;&#114;&#105;&#112;&#116;&gt;`
The HTML decode already handles this - maybe we should encode it like this by default (as opposed to `&lt;script&gt;`)
Views?

**Comment by balsamiqFlorian** [comment]
Yes, but to be honest. I would not mind to see both outputs. The HTML characters make an input text very long if you encode multiple times. Which can easily exceed a max length input.

**Comment by psiinon** [comment]
Would a checkbox on that dialog be ok (eg next to the 'HTML Encode' or would you want another text area. I'm just thinking that the Encode tab is getting a bit crowded already.
But I definitely think we should include this somehow, just a case of getting the UI right

**Comment by balsamiqFlorian** [comment]
Here is a quick potential preview.

![zap_encoder](https://cloud.githubusercontent.com/assets/3863260/12786670/5f27f008-ca92-11e5-8dbf-836a403e2902.png)
